src/python/pants/option/global_options.py

```diff
@@ -122,6 +122,12 @@ def register_bootstrap_options(cls, register):
     register('--pants-version', advanced=True, default=pants_version(),
              help='Use this pants version.')
 
+    # Like `--pants-version`, Pants code uses this only to verify that the user is using the requested interpreter
+    # version. It too may be grepped out of pants.ini to be used by things like setup scripts,
+    # runner scripts, IDE plugins, etc. in order to select which interpreter to use.
+    register('--pants-engine-python-version', advanced=True,
+             help='Use this Python version to run Pants. Valid values: 2.7, 3.6, or 3.7.')
```

Comment:
This shouldn't be an enum, because we don't know what values users will use for it. Anything that could legally be passed to `virtualenv`, essentially.

Reply:
I've gone back and forth on if we should enumerate all the supported versions here or not..

--

This question speaks to a related check that we're going to need to add: in `pants_loader.py` checking for a valid interpreter. This was suggested first by Kris here https://github.com/pantsbuild/pants/issues/6062#issuecomment-401983486. It will assert that they're using a supported Py version, i.e. not Py2.6 or 3.0-3.5 (soon 3.6+). 

The reason this belongs in `pants_loader.py` and not here in `global_options.py` or `options_initializer.py` is that we cannot make any assumptions about them using this option and how they're setting the interpreter to run Pants with. Even if they completely circumvent this option / our setup repo and try running Pants with something like Py3.4, we need to run the assertion.

--

I think we probably want to avoid, then, enumerating the options here, as that's the job of `pants_loader.py`.

Instead, my original intent was to give an example of what this value is supposed to look like, i.e. that it should be the major and minor number like `2.7` or `3.6`. We need it to be like this because of the check here: https://github.com/pantsbuild/pants/pull/7363/files#diff-c30fdc263a0a6d8522b90488e2a264d5R112.